cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,485 @@
+        Returns:
+            tf_idf: The resulting matrix giving a value (importance score)
+            for each word per topic
+            count: object of class CountVecWrapper
+        """
+        count = CountVecWrapper(ngram_range=ngram_range)
+        X = count.fit_transform(document_df)
+        multiplier = None
+
+        transformer = ClassTFIDF().fit(X, n_samples=m, multiplier=multiplier)
+
+        c_tf_idf = transformer.transform(X)
+
+        return c_tf_idf, count
+
+    def create_topics(self, docs_df):
+        """Extract topics from the clusters using a class-based TF-IDF
+        Arguments:
+            data: list with documents
+            cluster: HDBSCAN object
+        Returns:
+            tf_idf: The resulting matrix giving a value (importance score) for
+            each word per topic
+            count: object of class CountVecWrapper, which inherits from
+            CountVectorizer
+            docs_per_topics_topics: A list of unique topic labels
+            docs_df: cudf DataFrame created from data
+        """
+        docs_per_topics_topics = docs_df["Topic"].unique()
+
+        tf_idf, count = self.new_c_tf_idf(docs_df, len(docs_df))
+        return tf_idf, count, docs_per_topics_topics, docs_df
+
+    def top_n_idx_sparse(self, matrix, n):
+        """Return indices of top n values in each row of a sparse matrix
+        Retrieved from:
+            https://stackoverflow.com/questions/49207275/finding-the-top-n-values-in-a-row-of-a-scipy-sparse-matrix
+        Args:
+            matrix: The sparse matrix from which to get the
+            top n indices per row
+            n: The number of highest values to extract from each row
+        Returns:
+            indices: The top n indices per row
+        """
+        top_n_idx = []
+        mat_inptr_np_ar = matrix.indptr.get()
+        le_np = mat_inptr_np_ar[:-1]
+        ri_np = mat_inptr_np_ar[1:]
+
+        for le, ri in zip(le_np, ri_np):
+            le = le.item()
+            ri = ri.item()
+            n_row_pick = min(n, ri - le)
+            top_n_idx.append(
+                matrix.indices[
+                    le + cp.argpartition(
+                        matrix.data[le:ri], -n_row_pick
+                    )[-n_row_pick:]
+                ]
+            )
+        return cp.array(top_n_idx)
+
+    def top_n_values_sparse(self, matrix, indices):
+        """Return the top n values for each row in a sparse matrix
+        Args:
+            matrix: The sparse matrix from which to get the top n
+            indices per row
+            indices: The top n indices per row
+        Returns:
+            top_values: The top n scores per row
+        """
+        top_values = []
+        for row, values in enumerate(indices):
+            scores = cp.array(
+                [matrix[row, value] if value is not None
+                 else 0 for value in values]
+            )
+            top_values.append(scores)
+        return cp.array(top_values)
```

Comment:
We should push to a utils file . I suggest creating a utils folder and files like `utils/sparse_matrix_utils.py`